src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -540,6 +553,50 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 				}
 				break;
 
+			case IntrinsicId.Nullable_GetUnderlyingType:
+				foreach (var singlevalue in argumentValues[0].AsEnumerable ()) {
+					AddReturnValue (singlevalue switch {
+						SystemTypeValue systemType => systemType,
+						NullableSystemTypeValue nullableSystemType => new SystemTypeValue (nullableSystemType.UnderlyingTypeValue),
+						NullableValueWithDynamicallyAccessedMembers nullableDamValue => nullableDamValue.UnderlyingTypeValue,
+						ValueWithDynamicallyAccessedMembers damValue => damValue,
+						_ => GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes)
+					});
+				}
+				break;
+
+			case IntrinsicId.Type_MakeGenericType:
+				// Contains part of the analysis done in the Linker, but is not complete
+				// instanceValue here is like argumentValues[0] in linker
+				foreach (var value in instanceValue) {
+					// Nullables without a type argument are considered SystemTypeValues
+					if (value is SystemTypeValue typeValue
+							&& typeValue.RepresentedType.IsTypeOf ("System", "Nullable`1")) {
```

Comment:
nit: to reduce some of the nesting here, could you use a style that inverts the condition? Like this:
```csharp
if (!condition)
    continue;
foreach (var argumentValue in argumentValues[0]) {
// ...
}
```

And similar below, inside the for loop.